src/main.ts

```diff
@@ -572,9 +584,11 @@ function generateDecode(ctx: Context, fullName: string, messageDesc: DescriptorP
     } else if (isValueType(ctx, field)) {
       const type = basicTypeName(ctx, field, { keepValueType: true });
       readSnippet = code`${type}.decode(reader, reader.uint32()).value`;
-    } else if (isTimestamp(field)) {
+    } else if (isTimestamp(field) && options.useDate === DateOption.DATE) {
       const type = basicTypeName(ctx, field, { keepValueType: true });
       readSnippet = code`${utils.fromTimestamp}(${type}.decode(reader, reader.uint32()))`;
+    } else if (isTimestamp(field) && options.useDate === DateOption.STRING) {
+      readSnippet = code`reader.${toReaderCall(FieldDescriptorProto_Type.TYPE_STRING)}()`;
```

Comment:
I think this needs a utility like above to convert from the protobuf message Timestamp type to string. Maybe just modify the fromTimestamp function so that it returns a string if useDate===DateOption.String and just leave this code as it was.